### ISSUE TITLE
Production Rules does not work without else (non-testing) case 

### ISSUE THREAD

**Issue by kanitw** [vega/vega#1366]
Basically, if without else doesn't work. 
Here is an example where `fill` doesn't have an "else" case.
This will lead to `SyntaxError: Unexpected token ;` 

```
{
  "$schema": "https://vega.github.io/schema/vega/v4.json",
  "description": "A scatterplot showing horsepower and miles per gallons for various cars.",
  "autosize": "pad",
  "padding": 5,
  "width": 200,
  "height": 21,
  "style": "cell",
  "data": [
    {
      "name": "source_0",
      "url": "data/cars.json",
      "format": {"type": "json", "parse": {"Horsepower": "number"}}
    }
  ],
  "marks": [
    {
      "name": "marks",
      "type": "symbol",
      "style": ["point"],
      "from": {"data": "source_0"},
      "encode": {
        "update": {
          "opacity": {"value": 0.7},
          "fill": [
            {
              "test": "datum[\"Horsepower\"] === null || isNaN(datum[\"Horsepower\"])",
              "value": "none"
            }
          ],
          "stroke": [
            {
              "test": "datum[\"Horsepower\"] === null || isNaN(datum[\"Horsepower\"])",
              "value": "none"
            },
            {"value": "#4c78a8"}
          ],
          "tooltip": {
            "signal": "{\"Horsepower\": format(datum[\"Horsepower\"], \"\")}"
          },
          "x": {"scale": "x", "field": "Horsepower"},
          "y": {"signal": "height", "mult": 0.5}
        }
      }
    }
  ],
  "scales": [
    {
      "name": "x",
      "type": "linear",
      "domain": {"data": "source_0", "field": "Horsepower"},
      "range": [0, {"signal": "width"}],
      "nice": true,
      "zero": true
    }
  ],
  "axes": [
    {
      "scale": "x",
      "orient": "bottom",
      "grid": false,
      "title": "Horsepower",
      "labelFlush": true,
      "labelOverlap": true,
      "tickCount": {"signal": "ceil(width/40)"},
      "zindex": 1
    },
    {
      "scale": "x",
      "orient": "bottom",
      "grid": true,
      "tickCount": {"signal": "ceil(width/40)"},
      "domain": false,
      "labels": false,
      "maxExtent": 0,
      "minExtent": 0,
      "ticks": false,
      "zindex": 0
    }
  ],
  "config": {"axisY": {"minExtent": 30}}
}
```

**Comment by kanitw** [comment]
I see.  I thought this is optional as the [documentation](https://vega.github.io/vega/docs/marks/#production-rules) says that this else-clause "can be specified" (not "must be"). 

> A single value reference, without a test property, can be specified as the final element within the rule to serve as the else condition.

----

In any case, shouldn't it fall back to the default value defined in the config for that channel? 
If not, using null is probably better than throwing an error. 

If you disagree, we should at least show a nicer error than `SyntaxError: Unexpected token ;`.
Btw, making JSON schema validates this might be a bit tricky (I'm not sure if it is possible at all).

**Comment by jheer** [comment]
You're right about JSON schema: I don't see a good way to define this constraint.

The problem with providing a config value is that (a) configuration is applied _before_ encodings are processed, based on whether or not certain encodings are defined and (b) some of those configuration options (i.e., `stroke`/`fill`) can be dependent on combinations of encodings. I'm reticent to complicate the logic here.

I think for now I will make unterminated rule chains default to an else clause of `null`, plus issue a warning (if and when we add warning logging to the parser).

**Comment by jheer** [comment]
Fixed in vega-parser@3.7.0.
